test/terra/backends/test_parameterized_qobj.py

```diff
@@ -404,6 +404,29 @@ def test_run_empty(self):
         with self.assertRaises(AerError):
             res = backend.run(circuit, shots=shots, parameter_binds=parameter_binds).result()
 
+    def test_parameters_with_barrier(self):
+        """Test parameterized circuit path with barrier"""
+        shots = 2000
+        backend = AerSimulator()
+        circuit = QuantumCircuit(3)
+        theta = Parameter("theta")
+        phi = Parameter("phi")
+        circuit.rx(theta, 0)
+        circuit.rx(theta, 1)
+        circuit.rx(theta, 2)
+        circuit.barrier()
+        circuit.rx(phi, 0)
+        circuit.rx(phi, 1)
+        circuit.rx(phi, 2)
+        circuit.barrier()
+        circuit.measure_all()
+
+        parameter_binds = [{theta: [pi / 2], phi: [pi / 2]}]
+        res = backend.run([circuit], shots=1024, parameter_binds=parameter_binds).result()
```

Comment:
I think the `shots=2000` on line 409 is accidentally left in - it's never used.  (I'm a little surprised there's no linter that complains about it.)